### ISSUE TITLE
creating DynamicalSystem from Atoms resets masses

### ISSUE THREAD

**Issue by noambernstein** [libAtoms/QUIP#25]
If you have an Atoms object with masses that don't match the default for the atomic numbers, creating a DynamicalSystem from that atoms object resets the masses to those from PeriodicTable.f95 ElementMass[] for that atomic number. See the following script.

```
import quippy, sys, ase

at = quippy.Atoms(ase.io.read(sys.argv[1]))
at.write("stdout")

at.add_property('mass', 0.0)
at.mass[:] = at.get_masses()*quippy.MASSCONVERT

print "at.mass ", at.mass
ds=quippy.DynamicalSystem(at)
print "ds.atoms.mass ", ds.atoms.mass
```

Example input: 

```
2
Lattice="6.0 0.0 0.0     0.0 6.0 0.0     0.0 0.0 6.0" Properties=species:S:1:pos:R:3:momenta:R:3:force:R:3:masses:R:1
O 0.0 0.0 0.0 0.1 0.0 0.0     0.0 0.0 0.0   1.0
O 2.0 0.0 0.0 -0.1 0.0 0.0     0.0 0.0 0.0   1.0
```

Example output:
tin 1122 : ./test_mass.py t.extxyz 

```
2
cutoff=-1.00000000 nneightol=1.20000000 pbc="T T T" Lattice="6.00000000       0.00000000       0.00000000       0.00000000       6.00000000       0.00000000       0.00000000       0.00000000       6.00000000" Properties=species:S:1:pos:R:3:masses:R:1:momenta:R:3:force:R:3:Z:I:1
O               0.00000000      0.00000000      0.00000000      1.00000000      0.10000000      0.00000000      0.00000000      0.00000000      0.00000000      0.00000000       8
O               2.00000000      0.00000000      0.00000000      1.00000000     -0.10000000      0.00000000      0.00000000      0.00000000      0.00000000      0.00000000       8
at.mass  [ 103.64272176  103.64272176]
ds.atoms.mass  [ 1658.2213769  1658.2213769]
```

**Comment by noambernstein** [comment]
OK - I now understand where this is happening, but not why.  For reasons that I don't understand, the _second_ statement in the sequence

```
    ds = DynamicalSystem(at)
    print ds.atoms.mass
```

triggers a call to 

```
    quippy/atoms.py:Atoms.__init__
```

where, around line 415, 

```
    self.set_atoms(self.z)
```

is called.  This call, I believe, is the one that overrides the mass property.

I don't think I understand the logic of atoms.py enough to figure out the right way to fix this, but maybe someone else does?

**Comment by jameskermode** [comment]
Should now be fixed. I've added a new test to catch any regressions.

If you are interested in why it was the `print` statement that triggered the change, this is because the Python Atoms object that wraps `ds.atoms` is only constructed when first accessed. This is what leads to the call to `Atoms.__init__()` which then calls `set_atoms()`. I've changed it to pass the current mass along, but it would be better to avoid reinitialising if all we are doing is construcing a wrapper for an existing Atoms.
